stm32-modules/heater-shaker/firmware/heater_task/heater_hardware.c

```diff
@@ -284,6 +286,33 @@ void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc) {
     }
 }
 
+bool heater_hardware_set_offsets(struct writable_offsets* to_write) {
+    FLASH_EraseInitTypeDef pageToErase = {.TypeErase = FLASH_TYPEERASE_PAGES, .PageAddress = OFFSETS_PAGE_ADDRESS, .NbPages = 1};
+    uint32_t pageErrorPtr = 0; //pointer to variable  that contains the configuration information on faulty page in case of error
+    uint32_t ProgramAddress1 = OFFSETS_PAGE_ADDRESS + (offsetof(struct writable_offsets, const_b) * 8); //offsetof returns bytes
+    uint32_t ProgramAddress2 = OFFSETS_PAGE_ADDRESS + (offsetof(struct writable_offsets, const_c) * 8);
+    uint32_t ProgramAddress3 = OFFSETS_PAGE_ADDRESS + (offsetof(struct writable_offsets, const_flag) * 8);
+
+    HAL_StatusTypeDef status = HAL_FLASH_Unlock();
+    if (status == HAL_OK) {
+        status = HAL_FLASHEx_Erase(&pageToErase, &pageErrorPtr);
+        if (status == HAL_OK) {
+            status = HAL_FLASH_Program(FLASH_TYPEPROGRAM_DOUBLEWORD, ProgramAddress1, to_write->const_b);
+            status = HAL_FLASH_Program(FLASH_TYPEPROGRAM_DOUBLEWORD, ProgramAddress2, to_write->const_c);
+            status = HAL_FLASH_Program(FLASH_TYPEPROGRAM_DOUBLEWORD, ProgramAddress3, to_write->const_flag);
+            if (status == HAL_OK) {
+                status = HAL_FLASH_Lock();
+            }
+        }
+    }
+    return (status == HAL_OK);
+}
+
+uint64_t heater_hardware_get_offset(size_t addr_offset) {
+    uint32_t AddressToRead = OFFSETS_PAGE_ADDRESS + (addr_offset * 8); //addr_offset in bytes
+    return *(uint64_t*)AddressToRead;
```

Comment:
```suggestion
    return (uint64_t)AddressToRead;
```

Reply:
AddressToRead is flash register memory address, so here we can convert it to a pointer to match/indicate that and then dereferenced to access its data, as was done with serial number retrieval

Reply:
Oh yup I was misinterpreting that, sorry